### ISSUE TITLE
unable to call setDescription on RoutineInfo.Builder

### ISSUE THREAD

**Issue by hamnis** [googleapis/java-bigquery#3169]
#### Environment details
library version 2.38.0

#### Steps to reproduce

  1. try to setDescription on RoutineInfo.Builder
  2. fail compilation because package scope

#### Code example

```java
RoutineInfo.of(ROUTINE_ID)
    .toBuilder()
    // .setDescription("some description") //fails to compile
    .setLanguage(LANGUAGE)
    .setArguments(ARGUMENT_LIST)
    .setReturnType(RETURN_TYPE)
    .setImportedLibraries(IMPORTED_LIBRARIES)
    .setBody(BODY)
    .build();
```

**Comment by PhongChuong** [comment]
Hi @hamnis ,
It is currently possible to use the builder to setDescription. Did you want to setDescription on RoutineInfo directly?

**Comment by hamnis** [comment]
@PhongChuong it is not possible to use the builder since the `builder.setDescription(String)` is package protected, See the related PR.

**Comment by PhongChuong** [comment]
Looking Routine more carefully. It seems like the description is coming from BigQuery backend. I'm hesitant in making setDescription public as it allows users to override the backend description value which makes debugging backend issues harder in the future. Is there a reason to override the description coming from BigQuery? Would it be possible to bundle another description value with RoutineInfo in your code outside of RoutineInfo?

Let me know what you think,

**Comment by hamnis** [comment]
We are managing all the tables, views, materialized views, UDF and TFVs in our platform through our own scala library. 
So we have built our own internal migration platform, meaning that all displayable properties in BQ should be possible to set in the scala library.

As a workaround for it not being public, we have added a class that exposes that in the `com.google.cloud.bigquery` package. I find it unfortunate that we have to do that.

https://github.com/nrkno/bigquery-scala/blob/main/google-client/src/main/scala/com/google/cloud/bigquery/RoutineInfoHelper.scala
